python/cugraph/cugraph/experimental/compat/nx/algorithms/link_analysis/pagerank_alg.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cugraph
+import cugraph.utilities
+import cudf
+import numpy as np
+
+
+def cudify(d):
+    """
+    Converts a dictionary which is what networkx sends to a
+    cudf.DataFrame required in cugraph
+
+    Parameters
+    ----------
+    dictionary with node ids(key) and values
+
+    Returns
+    -------
+    a DataFrame of (vertex)ids and values.
+    """
+    if d is None:
+        return None
+    if not(isinstance(d, dict)):
+        raise TypeError("type_name must be a dict, got: "
+                        f"{type(d)}")
+    k = np.fromiter(d.keys(), dtype="int32")
```

Comment:
This can be a very expensive operation. `cudf.DataFrame` support dictionaries as input so you can directly pass it like this` df = cudf.DataFrame(d)` and rename the column name with the `.rename` function